File: tests/fixtures/java/DeepNesting.java
import java.util.List;

public class DeepNesting {
    public boolean deeplyNested(List<List<List<Integer>>> data) {
        for (var group : data) {
            if (!group.isEmpty()) {
                for (var items : group) {
                    if (!items.isEmpty()) {
                        for (var val : items) {
                            if (val > 0) {
                                process(val);
                            }
                        }
                    }
                }
            }
        }
        return true;
    }

    public boolean moderatelyNested(List<Integer> data) {
        for (var item : data) {
            if (item > 0) {
                process(item);
            }
        }
        return true;
    }

    private void process(int x) {}
}

File: tests/fixtures/c/deep_nesting.c
void deeply_nested(int* data, int n) {
    for (int g = 0; g < n; g++) {
        if (data[g] > 0) {
            for (int i = 0; i < n; i++) {
                if (data[i] > 0) {
                    for (int s = 0; s < n; s++) {
                        if (data[s] > 0) {
                            process(data[s]);
                        }
                    }
                }
            }
        }
    }
}

void moderately_nested(int* data, int n) {
    for (int i = 0; i < n; i++) {
        if (data[i] > 0) {
            process(data[i]);
        }
    }
}

void process(int x) {}
